resources/assets/lib/beatmap-discussions/editor.tsx

```diff
@@ -378,6 +385,28 @@ export default class Editor extends React.Component<Props, State> {
 
   serialize = () => serializeSlateDocument(this.state.value);
 
+  showConfirmationIfRequired = () => {
+    const docContainsProblem = slateDocumentContainsProblem(this.state.value);
+    const canDisqualify = currentUser.is_admin || currentUser.is_moderator || currentUser.is_full_bn;
+    const willDisqualify = this.props.beatmapset.status === 'qualified' && docContainsProblem;
+    const canReset = currentUser.is_admin || currentUser.is_nat || currentUser.is_bng;
+    const willReset =
+      this.props.beatmapset.status === 'pending' &&
+      this.props.beatmapset.nominations && this.props.beatmapset.nominations.current > 0 &&
+      docContainsProblem;
+
+    let confirmed = true;
+
+    if (canDisqualify && willDisqualify) {
+      confirmed = confirm(osu.trans('beatmaps.nominations.reset_confirm.disqualify'));
```

Comment:
skip `confirmed` variable and return immediately? Also in a quick glance it looks like there might be up to two confirmations (except it's not actually possible).